### ISSUE TITLE
"Invalid Add-on repository" error

### ISSUE THREAD

**Issue by munnecke** [magico13/ha-emporia-vue#69]
when I add the repository https://github.com/magico13/ha-emporia-vue to my home assistant web page, I get "Invalid Add-on repository" error... any way to fix this?

**Comment by BluGeni** [comment]
Might because you dont need to do custom repository. Its in the store now.

![image](https://user-images.githubusercontent.com/2058799/131696386-7fb9eb04-be6b-4852-96d4-d6c655b90f9c.png)

**Comment by magico13** [comment]
Hmm, I didn't do a PR to get it into the default store yet so I would be surprised if it was in there. Might have to do a bit of investigation, maybe someone else had it added?

**Comment by munnecke** [comment]
Thanks for your responses.  I don't see it in my add on store.  I'm a
newcomer to HA.  I'm using the OS version on a Pi 4, and have been having
problems with my internet gateway (Node Red, Mosquitto, and others
returning bad gateway).  I can't figure out how to get to a command line,
or Samba access to my file structures.  So I hit a wall trying to install
HACS.  I'm happy to wait for it to come to the store for direct download.
probably should find a place to help me with my communications problems, as
well.

**Comment by magico13** [comment]
Unfortunately it's unlikely to end up in the addon store, just remain in HACS. There is a possibility that I will be getting an initially stripped down version into the Core though so you could wait for that. I run with a docker container, not a pi, but I'm sure there are people in the Home Assistant forum who can help you get access to the file system on the pi if that's something you're interested in.

**Comment by munnecke** [comment]
By the way, I'm noodling some ideas about integrating my grid/solar energy usage, HVAC, and air filtration to optimize comfort level (temp, air quality, energy costs per time of use) looks like HA is a great platform to work on. 

One suggestion for the Emporia energy management would be to add time of use electricity costs.  My grid provider (SDGE) has a special Electric Vehicle Time of Use fee that hugely favors off-peak charging ($.09/kwh) (for my two EVs), and rewards me for selling back solar energy during peak hours (up to $.56/kwh).  So while I'm drawing a lot off peak, and my net kwh usage for the month is high, it is offset by my sale of solar power at peak rates to the point that my bill is a credit for the year.  These rates vary by day of the week, holidays, season, etc. so it's a very complicated issue.  And they have baseline fixed costs (my rate costs $16/mo, for example).  So optimizing energy costs (not just kwh) becomes a really important part of energy management. And solar battery and EV charging makes it all the more interesting.  I don't know the HA archtitecture that well, but it seems to me that the cost issue should be a layer above the energy usage meters.  I'd be happy for any pointers within the HA community for these issues.

**Comment by magico13** [comment]
The energy dashboard that was added to home assistant in the last major update is a step towards that goal. They've got cost calculation with time of use support (though that can be a challenge to set up). You might also be able to use template sensors to take the energy data and transform it into costs yourself. Then you can use those cost sensors in automations however you like, for example alerting when it's high or raising the HVAC temperature one or two degrees during high ToU.

The energy dashboard is still pretty new and doesn't seem to calculate quite correctly yet so I expect it will have improvements in the coming months.

**Comment by munnecke** [comment]
I just got notice of Emporia version 3 release, which incorporates the time
of use costing I was just talking about.  I named by provider and rate
(SDGE/EV TOU 5) and it looks like it is working well.   Would these cost
numbers come through the API?

Also, how hard would it be to just read the meter directly through Zigbee?

**Comment by magico13** [comment]
I'm going to be updating the API library with some of the new data but getting that then into Home Assistant is more nuanced.

Unfortunately it's currently mostly impossible to get data directly from the Vue locally though there's some hope that they might support that option in the future.

**Comment by declan727** [comment]
Other custom Repositories has worked and the only thing I can see is they named their file repository.json instead of hacs.json in your files. I was just wondering.

**Comment by merlinacious** [comment]
@magico13 - I am still seeing this similar issue when trying to add your github repo in the HACS Custom Add-On. See screenshot here: https://imgur.com/a/KNUeTtl

**Comment by magico13** [comment]
> @magico13 - I am still seeing this similar issue when trying to add your github repo in the HACS Custom Add-On. See screenshot here: https://imgur.com/a/KNUeTtl

This integration has to be added via HACS, the Home Assistant Community Store, which is a custom component. It is explicitly NOT an addon, that's a different thing from components. https://hacs.xyz/docs/setup/download/

This particular issue looks like it probably should be closed out so I'm going to do that.

**Comment by jumanjii1** [comment]
Repository 'magico13/ha-emporia-vue' exists in the store. This is what I get when installing Emporia-vue integration thru HACS. So I look for it in the HACS Store and it's not there. I tried other integrations and they all do the same thing or it hangs downloading. I GIVE UP

**Comment by magico13** [comment]
@jumanjii1 You have to add it via HACS as a custom repository. It is not an addon. If you're on a page in Home Assistant that mentions addons then you're in the wrong spot. Got to HACS, then integrations, then select the option to add custom repositories.
![image](https://github.com/magico13/ha-emporia-vue/assets/5289685/01eaa571-06e3-483b-b12f-40a0cec03c5a)

**Comment by magico13** [comment]
If you're getting "already in store" you may have added the custom repo. Once it's added to the store you have to search again in the main page of the HACS store (or find it in the recently added). Once you do that to actually install it then you have to go into the integrations list in Home Assistant to log in and set it up.

To put it another way, 1. Add the custom repo to the HACS store, 2. Install it in the HACS store, 3. Set it up in Home Assistant.

**Comment by jumanjii1** [comment]
Nope, there’s nothing there. I’m installing HA to another SD card now and am booting HA and trying again. I did have a Bluetooth dongle plugged into the Raspberry Pi so I removed it to see if that helps.
Thanks for your help


Sent from Yahoo Mail for iPhone


On Tuesday, September 5, 2023, 10:41 AM, Mike Marvin ***@***.***> wrote:




If you're getting "already in store" you may have added the custom repo. Once it's added to the store you have to search again in the main page of the HACS store (or find it in the recently added). Once you do that to actually install it then you have to go into the integrations list in Home Assistant to log in and set it up.

To put it another way, 1. Add the custom repo to the HACS store, 2. Install it in the HACS store, 3. Set it up in Home Assistant.
